src/block.rs

```diff
@@ -156,6 +156,17 @@ impl<T: Serialize + Clone> Block<T> {
         }
     }
 
+    /// Create an iterator over all proofs and transform into votes.
+    pub fn votes_iter<'a>(&'a self) -> Box<Iterator<Item = (&PublicInfo, Vote<T>)> + 'a> {
+        let payload = self.payload.clone();
+        Box::new(self.proofs.iter().map(move |proof| {
+            (
+                proof.node_info(),
+                Vote::compose(payload.clone(), *proof.sig()),
```

Comment:
Can't we just use `self.payload.clone()` here rather than taking another copy outside the loop?

Reply:
will that `move` the whole `self` ?

Reply:
No, I don't think so - we only have a reference to `self` there.